packages/ContentAutotagging/Engine/generic/process.types.ts

```diff
@@ -0,0 +1,57 @@
+export class ProcessRunParams {
+    sourceID: string;
+    startTime: Date;
+    endTime: Date;
+    numItemsProcessed: number;
+}
+
+export class ContentItemProcessParams {
+    text: string;
+    modelID: string;
+    minTags: number;
+    maxTags: number;
+    contentItemID: string;
+    contentTypeID: string;
+    contentFileTypeID: string;
+    contentSourceTypeID: string;
+}
+
+export class ContentItemProcessResults {
+    title: string;
+    author: string[];
+    publicationDate: Date;
+    keywords: string[];
+    content_text: string;
+    processStartTime: Date;
+    processEndTime: Date;
+    contentItemID: string;
+}
+
+export interface JsonObject {
+    [key: string]: any;
+}
+
+export const ModelTokenLimits: Record<string, number> = {
```

Comment:
@nico-oz @cadam11  - sounds like something we need to add to the `AI Models` entity? 

If agreed, create an issue @nico-oz  to do this and we can knock this out on your return to add the column to the entity, update existing rows for this attribute, and then come back here ad kill this data structure off and use the AI Model metadata from the `AIEngine` class